aten/src/ATen/native/cuda/DilatedMaxPool2d.cu

```diff
@@ -452,8 +452,11 @@ const Tensor& indices) {
   const int dilationH = safe_downcast<int, int64_t>(dilation[0]);
   const int dilationW = dilation.size() == 1 ? dilationH : safe_downcast<int, int64_t>(dilation[1]);
 
-  const auto memory_format = input_.suggest_memory_format();
-
+  auto memory_format = input_.suggest_memory_format();              
+  if (memory_format == MemoryFormat::Contiguous && input_.numel() > static_cast<int64_t>(std::numeric_limits<int>::max())) {
```

Comment:
Numeric limit should be defined based on int32 or int64 right? Fixed type widths here are important.